zitadel/v2/domain/funcs.go

```diff
@@ -25,9 +26,25 @@ func delete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	domainName := d.Id()
+	if d.Get(isPrimaryVar).(bool) {
+		resp, err := client.ListOrgDomains(ctx, &management.ListOrgDomainsRequest{})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		for _, domain := range resp.Result {
+			parts := strings.Split(clientinfo.Domain, ":")
+			if domain.IsVerified && domain.DomainName != domainName && strings.HasSuffix(domain.GetDomainName(), parts[0]) {
+				if _, err := client.SetPrimaryOrgDomain(ctx, &management.SetPrimaryOrgDomainRequest{Domain: domain.DomainName}); err != nil {
+					return diag.FromErr(err)
+				}
+				break
+			}
+		}
+	}
```

Comment:
Let's use the same logic in the update function when the `isPrimaryVar` property switches to false?

Reply:
that doesnt work, as the change function only works like that as there always has to be a primary domain